packages/flutter_tools/lib/src/doctor.dart

```diff
@@ -26,15 +26,20 @@ import 'vscode/vscode_validator.dart';
 
 Doctor get doctor => context[Doctor];
 
-class ValidatorTask {
-  ValidatorTask(this.validator, this.result);
-  final DoctorValidator validator;
-  final Future<ValidationResult> result;
+abstract class DoctorValidatorsProvider {
+  /// The singleton instance, pulled from the [AppContext].
+  static DoctorValidatorsProvider get instance => context[DoctorValidatorsProvider];
+
+  static final DoctorValidatorsProvider _defaultInstance = new _DefaultDoctorValidatorsProvider();
```

Comment:
Just make this static final member public and avoid the getter

Reply:
Done.